nx/test/nx/defn/evaluator_test.exs

```diff
@@ -321,6 +321,18 @@ defmodule Nx.Defn.EvaluatorTest do
              """
   end
 
+  defn mapped_inspect(a, b), do: print_value(a + b, fn x -> x * 2 end)
```

Comment:
I found this behavior a bit confusing. I expected the print to show the modified version but not actually affect the results.

Otherwise, what's the difference between this and just modifying the results before piping into print_value?

Reply:
It does not affect the result, the assertion is on the printed value :)

Reply:
I will add an assertion on the result too!

Reply:
Ah cool!

Reply:
I misread the test :)

Reply:
This is a great addition then!